\title{RADseq Works in Primates, Dammit.}
\author{
        Christina M. Bergey
			\and
			Andrew S. Burrell
			\and
			Luca S.J. Pozzi
			\and
			Todd, I suppose
	}
\date{\today}

\documentclass[12pt]{article}

\usepackage{color}

\begin{document}
\maketitle

\begin{abstract}
\ldots Blah, blah, blah, RADseq, blah, blah, Cercopithecoidea. \ldots
\end{abstract}

\section{Introduction}
\begin{itemize}
	\item Next-gen sequencing revolution promises gains in primatology
	\item Still expensive
	\item Many genomes, but still tough doing genomics on non-model organisms
	\item What is RADseq?
	\item Previous RADseq studies
	\item Why would it be good for primates
	\item PRESENT STUDY
	\begin{itemize}
		\item We did RADseq on 6 Cercopithecoids
		\item Assessed how well it worked
		\item Show it has promise for primates
	\end{itemize}
\end{itemize}

\section{Methods}

\paragraph{Library Preparation and Sequencing}

	Genomic DNA from 6 animals was digested with \emph{PspXI} (New England Biolabs) and used to create a multiplexed RAD tag library. Our library preparation method followed that of Etter et al, 2011 with the following modifications: the P1 adapter top\textcolor{red}{(?)} oligonucleotide was modified to have an overhang corresponding to the cut site of \emph{PspXI}, and a longer P2 adapter suitable for paired end sequencing was used (P2\_top: 5'-\textcolor{red}{SEQUENCEHERE}-3'; P2\_bottom: 5'-\textcolor{red}{SEQUENCEHERE}-3'). Individual-specific barcodes contained in the P1 adapter differed by at least \textcolor{red}{three} nucleotides. We chose \emph{PspXI} based on the results of \emph{in silico} digestion of the human, rhesus macaque, and baboon reference genomes using custom Perl scripts (\textcolor{red}{refs}). We sequenced the prepared library as one 150-cycle paired-end run of an Illumina MiSeq at the NYU Langone Medical Center's Genome Technology Center using a spike-in of 30\% PhiX DNA to control for low diversity in the library at the barcode and restriction sites. Sequences are available to download from the NCBI Short Read Archive (accession number \textcolor{red}{SRAXXXXXX.X}).

\paragraph{Sequence Analysis}

	Sequence reads were demultiplexed, or separated by barcode, and reads without an expected barcode or an intact restriction enzyme cut site were excluded from the analysis. Reads were then aligned to the rhesus macaque reference genome (v.1.0, Mmul_051212/rheMac2, \textcolor{red}{ref}) using BWA with default parameters. Reads that were unmapped, unpaired, duplicates, or that had low mapping quality were removed after alignment using Picard (\textcolor{red}{ref}) and BamTools (\textcolor{red}{ref}). 

	After performing local realignment around indels with GATK (\textcolor{red}{ref}), SNPs and short indels were identified using SAMtools mpileup and BCFtools (\textcolor{red}{ref}). A minimum coverage of 3 reads and a maximum of 100 was required to call a SNP or an indel at a given location.
	
	To analyze the degree of overlap between multiplexed individual's datasets, we...

\paragraph{Analysis Pipeline - Analysis of Degree of Overlap}
\begin{itemize}
	\item Calculate coverage of restriction site-associated regions
	\begin{itemize}
		\item Info on targeted intervals
		\begin{itemize}
			\item Total number possible targets in rhesus genome (compare to human too?)
			\item Total possible target BP
		\end{itemize}
		\item How many targets did we hit?
		\begin{itemize}
			\item BEDtools multiBamCoverage for this job
			\item Number and percentage of targets with coverage $\ge 1$
			\item Number and percentage of targets with coverage $\ge N$
		\end{itemize}
	\end{itemize}
	\item Count orthologous SNPs shared between individuals
	\begin{itemize}
		\item VCFtools vcf-compare for this job
	\end{itemize}
\end{itemize}

%\paragraph{Analysis Pipeline - Inferring Phylogeny}
%\begin{itemize}
%	\item Using Stacks? 
%	\item Using method like cichlid people?
%	\item Using method like Rubin et al %http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0033394
%\end{itemize}

\section{Results}
\begin{itemize}
	\item Info from analyzing reads with FastQC
	\begin{itemize}
		\item Number of reads (per ind. too see table)
		\item Total sequence bp (per ind. too see table)
		\item Maximum possible sequence depth (Cut?)
		\item Other stats that FastQC gives you?
	\end{itemize}
	\item Table: 
	\begin{itemize}
		\item Number of reads per animal
		\item Total sequenced bp per animal?
		\item Number that passed filtration
		\item Number of loci hit
		\item Number of loci hit with coverage $\ge N$
		\item Number of SNPs
	\end{itemize}
	\item SNP info from merged analysis
	\item SNP Venn diagram?
	\item Table of overlapping region, orthologous SNP counts
	%\item Phylogenetic tree
\end{itemize}

\section{Discussion}
\begin{itemize}
	\item RADseq is viable tool for researcher interested in primate phylogenetics, pop. gen.
	\item Enzyme choice allows control over coverage, number of individuals, number of loci.
	\item Potential problems with RADseq method
	\item Promise for primatology
\end{itemize}

\section{Acknowledgements}
Acknowledgements


\end{document}
